online-shop/src/app/services/cart.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+import { ShoppingCart } from 'src/shopping-cart';
+import { Product } from 'src/product';
+import { LocalStorageKeys } from 'src/local-storage-keys';
+import { Order } from 'src/order';
+import { Observable } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { backendURL } from 'src/utils';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class CartService {
+    constructor(private http: HttpClient) {}
+
+    cartList: ShoppingCart[] = this.initializeCart();
+
+    private initializeCart(): ShoppingCart[] {
+        const productsExistInLocalStorage = localStorage.getItem(LocalStorageKeys.shoppingCartKey);
```

Comment:
find a better name for this one, for me, it sounds like a boolean -> exists? true:false